develop/events.rst

```diff
@@ -112,3 +112,72 @@ Regular expression triggered events
 
 The :class:`supybot.callbacks.PluginRegexp` class provides some utilities
 for creating plugins that act on regular expression matching.
+
+Reacting to URLs
+----------------
+
+Using Limnoria's ``urlSnarfer`` will avoid loops with other bots and ignore private messages.
+
+.. code-block:: python
+
+    # An example plugin using urlSnarfer
+    # https://docs.limnoria.net/develop/commands.html#supybot.commands.urlSnarfer
+
+    from supybot import utils, plugins, ircutils, callbacks
+    from supybot.commands import *
+    from supybot.i18n import PluginInternationalization
+
+    _ = PluginInternationalization('ExampleSnarfer')
+
+    class ExampleSnarfer(callbacks.PluginRegexp):
+        """ Example URL snarfer """
+        # Note the class uses callbacks.PluginRegexp
+        # https://docs.limnoria.net/develop/callbacks.html#pluginregexp
+        
+        # Specify the handler method
+        regexps = ['snarfer_handler']
+        
+        # urlSnarfer() stops calling handlers once a message has been replied to
+        # Call this plugin before others that use urlSnarfer()
+        callBefore = ["Web"]
+        
+        @urlSnarfer
+        def snarfer_handler(self, irc, msg, match):
+            r'https://example\.com(/(\S*)|\s|$)'
+            # Messages that include a match to the regex are passed to this method
+            #  https://example.com
+            #  https://example.com/
+            #  https://example.com/anything/else/here 
+
+            channel = msg.channel
```

Comment:
These variables are only used once, so we can probably just do this inline in the registryValue lookup